### PR TITLE
feat(meson): add support for meson

### PR DESCRIPTION
**Pull request by Decodetalkers**
I write a treesitter for meson build, and I have tried it myself, I think the treesitter works good

### CODE REVIEW

File: queries/meson/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@punctuation.bracket
+
+[
+  "="
+  "=="
+  "!="
+  "and"
+  "+"
+  "+="
+  "not"
```
Comment (theHamsta) [inline queries/meson/highlights.scm:22]: `@keyword.operator`

File: queries/meson/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+[
+  "foreach"
+  "endforeach"
+  (keyword_break)
+  (keyword_continue)
+] @repeat
+
+;;; format
+(string) @string
+["@"] @keyword
```
Comment (theHamsta) [inline queries/meson/highlights.scm:41]: keyword?

File: queries/meson/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+(experession_statement
+	object: (identifier) @variable)
+
+(normal_command
+	command: (identifier) @function)
+
+variable: (identifier) @variable
```
Comment (theHamsta) [inline queries/meson/highlights.scm:49]: I think here is something missing

File: queries/meson/highlights.scm (hunk 2 above)
Comment (Decodetalkers) [reply queries/meson/highlights.scm:41]: > keyword?

So what is better?, it is this one
image

a format keyword

Comment (theHamsta) [reply queries/meson/highlights.scm:41]: But that's not a keyword in the sense of a reserved language keyword. Maybe you could use `@punctuation.special` like bash and python use for string interpolation?

File: queries/meson/highlights.scm (hunk 1 above)
Comment (theHamsta) [reply queries/meson/highlights.scm:22]: Same for `"and"`

File: queries/meson/highlights.scm (hunk 2 above)
Comment (Decodetalkers) [reply queries/meson/highlights.scm:41]: > But that's not a keyword in the sense of a reserved language keyword. Maybe you could use `@punctuation.special` like bash and python use for string interpolation?

Thanks

File: queries/meson/highlights.scm (hunk 1 above)
Comment (Decodetalkers) [reply queries/meson/highlights.scm:22]: > Same for `"and"`

What is same for and mean?

Comment (Decodetalkers) [reply queries/meson/highlights.scm:22]: friendly ping @@

Comment (theHamsta) [reply queries/meson/highlights.scm:22]: `@keyword.operator` for `"not"` and `"and"`. `@keyword` for all signs

Comment (theHamsta) [reply queries/meson/highlights.scm:22]: does it have `"or"`?

Comment (Decodetalkers) [reply queries/meson/highlights.scm:22]: > does it have `"or"`?

Thanks, it does have, I have forgotten it

**Comment by theHamsta** [comment]
Thank you!

**Review by theHamsta — APPROVED** [review]

**Comment by theHamsta** [comment]
You may still add  the standard `(comment) @comment` injection that many languages have.

**Comment by Decodetalkers** [comment]
> You may still add  the standard `(comment) @comment` injection that many languages have.

I will make a PR soon

**Pull request merged**